Breeze2

```diff
@@ -41,15 +49,35 @@ def save_config():
 
 if needs_installation():
     print(f"(Re)Installing Breeze's virtualenv in {BUILD_BREEZE_VENV_DIR}")
-    BUILD_BREEZE_VENV_DIR.mkdir(parents=True, exist_ok=True)
-    subprocess.run([sys.executable, "-m", "venv", f"{BUILD_BREEZE_VENV_DIR}"], check=True)
-    subprocess.run(
-        [f"{BUILD_BREEZE_VENV_PIP}", "install", "--upgrade", "-e", "."], cwd=BREEZE_SOURCE_PATH, check=True
-    )
+    try:
+        EnvBuilder(system_site_packages=False, upgrade=True, with_pip=True, prompt="breeze").create(
+            str(BUILD_BREEZE_VENV_DIR)
+        )
+    except Exception as e:
+        # in some cases (mis-configured python) the venv creation might not work via API
+        # (ensurepip missing). This is the case in case of default MacOS Python and Python executable
+        # Bundled in Windows executable, In this case we fallback to running venv as a tool using default
+        # Python3 found on path (in case of Windows Bundled exe, you don't even have a current
+        # interpreted executable available, because Python interpreter is executed through a library.
+        # and sys.executable points to the Bundled exe file.
+        BUILD_BREEZE_VENV_DIR.mkdir(parents=True, exist_ok=True)
+        subprocess.run(["python3", "-m", "venv", f"{BUILD_BREEZE_VENV_DIR}"], check=True, shell=True)
```

Comment:
Not convinced we should do this, but I’m not going to spend time arguing about it. I’m pretty confident you’ll regret this when people come complaining this failed on them (because `python3` points to a system Python for them and they didn’t install `python3-venv` or ensurepip).

Reply:
Also the `shell=True` part is unnecessary and only adds overhead.

Reply:
Is there a better option @uranusjr ?

Reply:
BTW. I was under the impression that as of 3.3 venv is part of the standard library? At least that's what https://docs.python.org/3.7/library/venv.html suggests. Am I wrong about it?

Reply:
(And from earielr looking at venv code, I believe it executes ensurepip under the hood).

Reply:
This is also what the documentation says:

> Unless the --without-pip option is given, ensurepip will be invoked to bootstrap pip into the virtual environment.

Reply:
Ah ok - I think you refer to this error : https://stackoverflow.com/questions/39539110/pyvenv-not-working-because-ensurepip-is-not-available - well, then I think the error message is rather clear:

```
The virtual environment was not created successfully because ensurepip is not
available.  On Debian/Ubuntu systems, you need to install the python3-venv
package using the following command.

    apt-get install python3-venv

You may need to use sudo with that command.  After installing the python3-venv
package, recreate your virtual environment.
```

Reply:
I think it's as good as it gets - there is no need to handle that in any other way, I think, if the error is that detailed and gives you that explanatory message, I think we should be good with that, but if you think there are other cases where the message will not be good, we can actually check it and write our own message - do you think there are other cases that the message might be misleading? I am all for making it as clear to our users as possible.

Reply:
BTW. shell is removed